pkg/controller/configmap/config.go

```diff
@@ -733,3 +812,86 @@ func IsMTUChanged(currConfigMap *corev1.ConfigMap, prevConfigMap *corev1.ConfigM
 		return true
 	}
 }
+
+func FillNsxAuthCfg(configmap *corev1.ConfigMap, nsxSecret *corev1.Secret) error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+	if err != nil {
+		log.Error(err, "failed to load ConfigMap")
+		return err
+	}
+
+	if nsx_cert, found := nsxSecret.Data["tls.crt"]; found {
+		if len(nsx_cert) == 0 {
+			log.Info(fmt.Sprintf("tls.crt is set as None in %s", operatortypes.NsxSecret))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "nsx_api_cert_file", "", true))
+		} else {
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "nsx_api_cert_file", "/etc/nsx-ujo/nsx-cert/tls.crt", true))
+		}
+	}
+
+	if nsx_key, found := nsxSecret.Data["tls.key"]; found {
+		if len(nsx_key) == 0 {
+			log.Info(fmt.Sprintf("tls.key is set as None in %s", operatortypes.NsxSecret))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "nsx_api_private_key_file", "", true))
+		} else {
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "nsx_api_private_key_file", "/etc/nsx-ujo/nsx-cert/tls.key", true))
+		}
+	}
+
+	if ca_file, found := nsxSecret.Data["tls.ca"]; found {
+		if len(ca_file) == 0 {
+			log.Info(fmt.Sprintf("tls.ca is set as None in %s", operatortypes.NsxSecret))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "ca_file", "", true))
+		} else {
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "ca_file", "/etc/nsx-ujo/nsx-cert/tls.ca", true))
+		}
+	}
+
+	// Write config back to ConfigMap data
+	(*data)[operatortypes.ConfigMapDataKey], err = iniWriteToString(cfg)
+	appendErrorIfNotNil(&errs, err)
+
+	if len(errs) > 0 {
+		return errors.Errorf("failed to fill nsx authentication configuration: %q", errs)
+	}
+	return nil
+}
+
+func FillLbCertCfg(configmap *corev1.ConfigMap, lbSecret *corev1.Secret) error {
+	errs := []error{}
+	data := &configmap.Data
+	cfg, err := ini.Load([]byte((*data)[operatortypes.ConfigMapDataKey]))
+	if err != nil {
+		log.Error(err, "failed to load ConfigMap")
+		return err
+	}
+
+	if lb_cert, found := lbSecret.Data["tls.crt"]; found {
+		if len(lb_cert) == 0 {
+			log.Info(fmt.Sprintf("tls.crt is set as None in %s", operatortypes.LbSecret))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "lb_default_cert_path", "", true))
+		} else {
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "lb_default_cert_path", "/etc/nsx-ujo/lb-cert/tls.crt", true))
+		}
+	}
+
+	if lb_key, found := lbSecret.Data["tls.key"]; found {
+		if len(lb_key) == 0 {
+			log.Info(fmt.Sprintf("tls.key is set as None in %s", operatortypes.LbSecret))
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "lb_priv_key_path", "", true))
+		} else {
+			appendErrorIfNotNil(&errs, fillDefault(cfg, "nsx_v3", "lb_priv_key_path", "/etc/nsx-ujo/lb-cert/tls.key", true))
```

Comment:
Put `"/etc/nsx-ujo/lb-cert/tls.key"` in defaults?

Reply:
I'm thinking the ncp will hit lb certification error if the tls.key is empty in the secret, however, we still set lb_priv_key_path as /etc/nsx-ujo/lb-cert/tls.key.